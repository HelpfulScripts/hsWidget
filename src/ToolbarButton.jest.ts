import * as widgets from './';
import m from "mithril";

const root = window.document.createElement("div");

const single = widgets.ToolbarButton.getSymbol('cross');
const multiple = [
    widgets.ToolbarButton.getSymbol('cross'),
    widgets.ToolbarButton.getSymbol('up')
];

describe('Toolbar', () => {
    beforeAll(()=>{
        m.mount(root, {view: () => m('', [
            m('.myPositioned', [
                 m(widgets.ToolbarButton, { symbols:single, onclick:()=>{} })
            ]),
            m('.myPositioned', [
                m(widgets.ToolbarButton, { symbols:multiple, onclick:()=>{} })
            ])
        ])});
    });
    it('creates', () => {
        expect(root).toMatchSnapshot();
    });
});
