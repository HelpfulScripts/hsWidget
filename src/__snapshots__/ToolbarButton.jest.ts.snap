// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar creates 1`] = `
<div>
  <div>
    <div
      class="myPositioned"
    >
      <div
        class="hs-corner-button"
      >
        ×
      </div>
    </div>
    <div
      class="myPositioned"
    >
      <div
        class="hs-corner-button"
      >
        ×
        ∧
      </div>
    </div>
  </div>
</div>
`;
